build-system/babel-plugins/babel-plugin-amd-custom-define/index.js

```diff
@@ -0,0 +1,61 @@
+const {readFileSync} = require('fs');
+const {join: pathJoin, relative} = require('path');
+
+const pathToModuleName = (filename) =>
+  filename.replace(/^(\.\/)?dist\//, '').replace(/(\.max)?\.m?js$/, '');
+
+let defineTemplate;
+
+module.exports = function (babel) {
+  const {types: t} = babel;
+
+  /** @return {string} */
+  function getTemplate() {
+    if (!defineTemplate) {
+      defineTemplate = readFileSync(
+        pathJoin(__dirname, 'define-template.js'),
+        'utf8'
+      );
+    }
+    return defineTemplate;
+  }
+
+  return {
+    name: 'amd-custom-define',
+    visitor: {
+      CallExpression(path, state) {
+        const callee = path.get('callee');
+        if (!callee.isIdentifier({name: 'define'})) {
+          return;
+        }
+        const [deps] = path.node.arguments;
+        if (!t.isArrayExpression(deps)) {
+          return;
+        }
+        const filename = relative(process.cwd(), state.filename);
+        for (const dep of deps.elements) {
+          if (!t.isLiteral(dep) || typeof dep.value !== 'string') {
```

Comment:
`t.isStringLiteral`, but I don't believe this can happen because ESM's import declarations are always static.